components/hab/src/command/studio.rs

```diff
@@ -93,24 +98,52 @@ mod inner {
         };
 
         if let Some(cmd) = find_command(command.to_string_lossy().as_ref()) {
-            try!(exec::exec_command(cmd, args));
+            try!(process::become_command(cmd, args));
         } else {
             return Err(Error::ExecCommandNotFound(command.to_string_lossy().into_owned()));
         }
         Ok(())
     }
+
+    pub fn rerun_with_sudo_if_needed(ui: &mut UI) -> Result<()> {
+        // If I have root permissions, early return, we are done.
+        if am_i_root() {
+            return Ok(());
+        }
+
+        // Otherwise we will try to re-run this program using `sudo`
+        match find_command(SUDO_CMD) {
+            Some(sudo_prog) => {
+                let mut args: Vec<OsString> =
+                    vec!["-p".into(), "[sudo hab-studio] password for %u: ".into(), "-E".into()];
+                args.append(&mut env::args_os().collect());
+                Ok(try!(process::become_command(sudo_prog, args)))
+            }
+            None => {
+                try!(ui.warn(format!("Could not find the `{}' command on the filesystem or in \
```

Comment:
I think the error here should read something like "Could not find the `{}` command. Is it in your PATH?". We should  always try and provide a way to fix the error along with what went wrong